Add changelog for 5.15.0 -> 5.16.0 (#332)
luanti-org/docs.luanti.org: content/about/changelog.md

Note that not all changes made to the code between releases are listed here. Fixes to bugs that were introduced after the previous release, small internal changes, code style fixes, and changes of the like are not listed.
If you want a list of _every_ change made between releases see the [commit log](https://github.com/luanti-org/luanti/commits/master).

> ## 5.15.2 → 5.16.0

Reviewer: ```suggestion
## 5.15.0 → 5.16.0
```

**[Release](https://github.com/luanti-org/luanti/releases/tag/5.16.0)**

Released on 8 May 2026.

### Deprecations and compatibility notes
- Writing to mod directories is now disallowed (deprecated in 5.9.0) (_sfan5_)

>   - Consider using the directory `core.get_mod_data_path()` (>= 5.9.0) instead.

Reviewer: ```suggestion
  - Global data can be stored in in `core.get_mod_data_path()` instead
```

### Client / Audiovisuals

> - Texture pack' `override.txt` now supports overriding overlay tiles (_DiogoAlves06_)

Reviewer: ```suggestion
- Texture pack `override.txt` now supports overriding overlay tiles (_DiogoAlves06_)
```
